### PR TITLE
Added v3 for go module

### PR DESCRIPTION
**Pull request by sayboras**
To add the `/v3` in the go mod as mentioned in https://github.com/buildkite/agent/issues/10.

I am not aware of releasing process, so didn't update CHANGELOG file.

### CODE REVIEW

**Comment by lox** [comment]
Thanks for this! 🙏🏻

File: clicommand/agent_start.go [hunk 1]
```diff
@@ -605,7 +605,7 @@ var AgentStartCommand = cli.Command{
 		}
 
 		l.Notice("Starting buildkite-agent v%s with PID: %s", agent.Version(), fmt.Sprintf("%d", os.Getpid()))
-		l.Notice("The agent source code can be found here: https://github.com/buildkite/agent")
+		l.Notice("The agent source code can be found here: https://github.com/buildkite/agent/v3")
```
Comment (sj26) [inline clicommand/agent_start.go:608]: Whoops!

File: process/process.go [hunk 2]
```diff
@@ -180,7 +180,7 @@ func (p *Process) Run() error {
 				// We can safely ignore this error, because
 				// it's just the PTY telling us that it closed
 				// successfully.  See:
-				// https://github.com/buildkite/agent/pull/34#issuecomment-46080419
+				// https://github.com/buildkite/agent/v3/pull/34#issuecomment-46080419
```
Comment (sj26) [inline process/process.go:183]: Whoops!

File: bootstrap/docker.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ func tearDownDeprecatedDockerIntegration(sh *shell.Shell) error {
 }
 
 // runDockerCommand executes a command inside a docker container that is built as needed
-// Ported from https://github.com/buildkite/agent/blob/2b8f1d569b659e07de346c0e3ae7090cb98e49ba/templates/bootstrap.sh#L439
+// Ported from https://github.com/buildkite/agent/v3/blob/2b8f1d569b659e07de346c0e3ae7090cb98e49ba/templates/bootstrap.sh#L439
```
Comment (sj26) [inline bootstrap/docker.go:85]: Whoops!

File: bootstrap/docker.go [hunk 4]
```diff
@@ -110,7 +110,7 @@ func runDockerCommand(sh *shell.Shell, cmd []string) error {
 }
 
 // runDockerComposeCommand executes a command with docker-compose
-// Ported from https://github.com/buildkite/agent/blob/2b8f1d569b659e07de346c0e3ae7090cb98e49ba/templates/bootstrap.sh#L462
+// Ported from https://github.com/buildkite/agent/v3/blob/2b8f1d569b659e07de346c0e3ae7090cb98e49ba/templates/bootstrap.sh#L462
```
Comment (sj26) [inline bootstrap/docker.go:113]: Whoops!

File: agent/artifact_uploader.go [hunk 5]
```diff
@@ -130,7 +130,7 @@ func (a *ArtifactUploader) Collect() (artifacts []*api.Artifact, err error) {
 			// If a glob is absolute, we need to make it relative to the root so that
 			// it can be combined with the download destination to make a valid path.
 			// This is possibly weird and crazy, this logic dates back to
-			// https://github.com/buildkite/agent/commit/8ae46d975aa60d1ae0e2cc0bff7a43d3bf960935
+			// https://github.com/buildkite/agent/v3/commit/8ae46d975aa60d1ae0e2cc0bff7a43d3bf960935
```
Comment (sj26) [inline agent/artifact_uploader.go:133]: Whoops!

**Pull request merged**